### DISCUSSION TITLE
Ryot and fitness tracking

### DISCUSSION THREAD

**Discussion by IgnisDa** [IgnisDa/ryot#235] [General]
Now that most of the media tracker features have been implemented, I'd like to gather some community feedback on what all should Ryot support for its fitness capabilities.

- I'd like to create a separate Expo application that has an interface like the [Strong](https://play.google.com/store/apps/details?id=io.strongapp.strong) app. This application will not be embedded in the ryot binary and just sync with the tracking server. 
- Support imports via a defined interface.

Eventually I'd also like Ryot to have health tracking (food, medicines etc), but that is further down the line. This should probably have some integration with third party self hosted apps that could do this better.

I would like the following additional capabilities:

- [x] Measuring body weight, height etc.
- [ ] Allow taking pictures for progress

**Comment by edersong** [comment]
For Health, I would request a way to track the following:
- [x] Body measures: weight, height, arm, leg, waist, chest, fat percentage etc.
- [ ] Medicines usage
- [ ] Symptoms
- [ ] Blood pressure
- [ ] Workouts (schedule plan and done)
- [ ] Upload pictures to track progresses
- [ ] Upload health exams results

With so many features request, I think that a separated a app for health tracking will be better due to infinite ways to grow. ;-)
There is a lack of a good selfhosted docker based Health app that can be developed using the excellent base of Ryot ;-)

**Reply by IgnisDa** [reply]
@edersong Wdym by schedule and plan workouts?

**Reply by IgnisDa** [reply]
I don't get it

**Reply by edersong** [reply]
I mean, to plan a workout before it with the exercises that we intend to do and scheduling it for a date.
The case is:
Prepare all the execises to be performed before go to the gym and set the dates that I'm intend to perform the same set of exercises, so grouping exercises like a "collection" and set dates in advance to perform it.

**Reply by IgnisDa** [reply]
Ah you're talking about workout templates/schedules?

Please create a new issue for it.

**Comment by IgnisDa** [comment]
Thanks for the suggestions, i agree with all of them!

> With so many features request, I think that a separated a app for health tracking will be better

So what i plan to do further the line is make a separate app (and website) for each facet of Ryot, eg fitness, media etc.

**Comment by Horib** [comment]
Is it possible to implement a base Ryot App without any features as in Ryot Core and then add the different "facets" such as fitness, media as modules or plugins?

With this approach, Ryot could extend their Core API and integrate both fitness, media and i.e. a workout tracker separately.
And having well defined API contracts between the Core and its Modules/plugins makes everything more maintainable.

Just a suggestion

**Comment by IgnisDa** [comment]
I'm not really sure what you mean by a base app, but since this is a monorepo, we can easily share base components (or whatever) across apps.

Or are you talking about a single Ryot app with "features" (like media, fitness etc) which can be enabled or disabled per user?

**Comment by IgnisDa** [comment]
I was looking into the exercise databases i can use. I could find https://github.com/yuhonas/free-exercise-db and https://rapidapi.com/justin-WFnsXH_t6/api/exercisedb/. The first one is free but doesn't have good illustrations. The second one needs an API key (and would have to be supplied by the user since there is a hard limit), but has really good illustrations.

Which one would people prefer?

EDIT: I should clarify that API limits would not be a problem on a per instance basis since data would be cached anyway. But i cannot hardcode my API Key.

**Comment by nodecentral** [comment]
Here’s a good one..

https://github.com/wger-project/wger

**Comment by IgnisDa** [comment]
For everyone following this discussion, the workout tracking feature of Ryot is working now!

**Reply by edersong** [reply]
Awesome!
New feature suggestion: 
**Integration with Google Fit**  to have all exercises and measurements from there locally available and synchronized.

**Reply by IgnisDa** [reply]
Yep. Please make an issue for this. I will work on it later.

**Reply by edersong** [reply]
done!

**Reply by filcuk** [reply]
I just want to say, the way you implemented this is fantastic.  
I've tried wger, but I don't want to plan the workouts, I just want to pick up the weights, start tracking and go.  
Genuinely great work.

**Reply by IgnisDa** [reply]
Thanks! I have been using Strong App for years now so i just used the same UI and added improvements.

**Comment by civilblur** [comment]
In addition to "exercises", would it be relevant to add "routines"? As in, each routine entry would contain:
- Title
  - The name of the routine.
  - When starting a new workout, the workout title will get prepopulated with the routine name.
- Multiple exercises
  - With option to allow users to predefine each exercise's sets/reps/measurements 

This concept can be found in the Android app [GymRoutines](https://f-droid.org/en/packages/com.noahjutz.gymroutines/), and for me, it saves a lot of time, and requires less interaction with my phone when I workout. It also helps label each exercise with the routine title, instead of e.g. the default "Morning routine".

I believe it's slightly different from what [edersong suggested](https://github.com/IgnisDa/ryot/discussions/235#discussioncomment-6630663).

Edit: Spelling, phrasing

**Reply by IgnisDa** [reply]
Is this similar to #461?

**Reply by civilblur** [reply]
After reading the discussion, in #461, it indeed sounds similar. The terms "template" and "routine" are essentially interchangeable.

It might have been implied already, but if not, the one thing that hasn't been mentioned is how it would be nice if the routine title would automatically populate the workout session title.

In #461, the "Collection" page was mentioned as a workaround. Currently, that page return "Application Error" for me, so I didn't realize there was a workaround. Will see if I can fix my setup so I can explore that part. **Edit:** Seems like it's related to ad blocker.

**Reply by IgnisDa** [reply]
Please mention what you expect from the feature on the issue itself so that I can reference it while I'm implementing it.

As for your error, if you're able to fix it, great. Otherwise open a new issue.

**Reply by civilblur** [reply]
I've added my suggestion to #461, I'll take a closer look at the error I get and see if it is applicable from on a broader scale (opposed to just me) before creating an issue.

**Reply by civilblur** [reply]
> In #461, the "Collection" page was mentioned as a workaround. Currently, that page return "Application Error" for me, so I didn't realize there was a workaround. Will see if I can fix my setup so I can explore that part. **Edit:** Seems like it's related to ad blocker.

Found out the reason, this was a "me problem" and a very specific case.

I've set my reverse proxy to use the subdomain `track`, and the ad blocker was matching the filter rule `://track.*/collect`, i.e. `track.domain.tld/collect` `ions/list/yours`.
